lib/rspec/matchers.rb

```diff
@@ -698,7 +698,11 @@ def all(expected)
     #       matchers (due to how the custom matcher DSL was evaluated in 2.x,
     #       `match` could not be used there), but is no longer needed in 3.x.
     def match(expected)
-      BuiltIn::Match.new(expected)
+      if ::String === expected
+        BuiltIn::Captures.new(expected)
+      else
+        BuiltIn::Match.new(expected)
+      end
```

Comment:
no, this refers to the opposite case to the one now, where the regex is the target and the string is the value passed into the matcher.